src/main/java/org/dataconservancy/packaging/shared/IpmPackager.java

```diff
@@ -0,0 +1,173 @@
+import org.dataconservancy.packaging.tool.model.GeneralParameterNames;
+import org.dataconservancy.packaging.tool.model.PackageGenerationParameters;
+import org.dataconservancy.packaging.tool.model.PackageState;
+import org.dataconservancy.packaging.tool.model.ParametersBuildException;
+import org.dataconservancy.packaging.tool.model.PropertiesConfigurationParametersBuilder;
+import org.dataconservancy.packaging.tool.model.RDFTransformException;
+import org.dataconservancy.packaging.tool.model.ipm.Node;
+
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Creates a DC package from a content provider, metadata and generation parameters.
+ * Created by wbt on 12/01/16.
+ */
+public class IpmPackager {
+
+    // TODO - There is a GitHub issue in osf-packaging about this use of this system property.
+    private String packageLocation = System.getProperty("java.io.tmpdir");
+    private String packageName = "MyPackage";
+
+    private final ClassPathXmlApplicationContext cxt = new ClassPathXmlApplicationContext(
+        "classpath*:org/dataconservancy/config/applicationContext.xml",
+        "classpath*:org/dataconservancy/packaging/tool/ser/config/applicationContext.xml");
+
+    /**
+     * API call to create an instance of a packager.  The name and location of the packages
+     * created by the packager can be changed before the package is built.
+     */
+    public IpmPackager() {
+    }
+
+    /**
+     * API call to change the file location where a package will be created.
+     * The location defaults to Java's temp IO folder.
+     * @param packageLocation The existing folder where the package file will be created.
+     */
+    public void setPackageLocation(String packageLocation) {
+        this.packageLocation = packageLocation;
+    }
+
+    /**
+     * API call to change the file name of the package that will be created.
+     * The name defaults to "MyPackage".
+     * @param packageName The root name of the package folder that will be created.
+     */
+    public void setPackageName(String packageName) {
+        this.packageName = packageName;
+    }
+
+    /**
+     * API call for creating a package from the given content, metadata
+     * and generation parameters.
+     * @param contentProvider Source of domain objects and IpmTree content
+     * @param metadataStream - A stream to metadata, as Java Properties.
+     * @param paramsStream A stream to package generation parameters, as Java Properties.
+     * @return A populated Package
+     * @throws RuntimeException if the content cannot be processed successfully.
+     */
+    public Package buildPackage(AbstractContentProvider contentProvider,
+                                InputStream metadataStream,
+                                InputStream paramsStream) {
+        try {
```

Comment:
try-with-resources using `contentProvider`?  Or is the caller responsible for calling `close()`?  Either way, the javadoc should be clear about who is responsible for calling `close()`